### ISSUE TITLE
[Bug] get_actor right after ray.kill returns the dead actor.

### ISSUE THREAD

**Issue by rkooo567** [ray-project/ray#20129]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Core

### What happened + What you expected to happen

Currently, calling `get_actor` right after `ray.kill` can return the stale dead actor. Since ray.kill is blocking, and it is passing through GCS, it should theoretically never happen, so there must be a bug. 

### Versions / Dependencies

master

### Reproduction script

https://github.com/ray-project/ray/blob/f65cc72b4ccb43e13c04f1273248f4f86ed85cae/python/ray/tests/test_actor_advanced.py#L1327

Basically, we should fix this. 

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!
